### ISSUE TITLE
Kubelet extra args not parsed correctly for controller nodes

### ISSUE THREAD

**Issue by puetzp** [puppetlabs/puppetlabs-kubernetes#260]
## What you expected to happen?
Init-/JoinConfiguration sections in config.yaml should be populated with extra args as specified in kubelet_extra_arguments parameter.
## What happened?
Including kubernetes class in a manifest like so:
```puppet
class { 'kubernetes':
...
  controller => true,
  kubelet_extra_arguments => [ "node_ip: ${facts[networking][ip]}" ],
...
}
```
does not create the expected configuration parts in config.yaml, which should be:
```yaml
nodeRegistration:
  name: <name>
  kubeletExtraArgs:
    node_ip: <ip>
```
Instead, kubeletExtraArgs are missing and ```kubectl get nodes``` reports the wrong IP address (different NIC).

Creating a worker node using the same settings however leads to the expected result.
Looking at the code:
```
 if $cloud_provider {
    $cloud_args = $cloud_config ? {
      undef   => ["cloud-provider: ${cloud_provider}"],
      default => ["cloud-provider: ${cloud_provider}", "cloud-config: ${cloud_config}"],
    }
    $apiserver_merged_extra_arguments = concat($apiserver_extra_arguments, $cloud_args)
    $kubelet_merged_extra_arguments = concat($kubelet_extra_arguments, $cloud_args)
    $controllermanager_merged_extra_arguments = $cloud_args

    # could check against Kubernetes 1.10 here, but that uses alpha1 config which doesn't have these options
    if $cloud_config {
      # The cloud config must be mounted into the apiserver and controllermanager containers
      $controllermanager_extra_volumes = $apiserver_extra_volumes = {
        'cloud' => {
          hostPath  => $cloud_config,
          mountPath => $cloud_config,
        }
      }
    }
  }
  else {
    $apiserver_merged_extra_arguments = $apiserver_extra_arguments
    $apiserver_extra_volumes = {}
    $controllermanager_merged_extra_arguments = []
    $controllermanager_extra_volumes = {}
}
```
there should be a ```$kubelet_merged_extra_arguments = $kubelet_extra_arguments``` in the else-block like you do for worker nodes:
```
  if !empty($cloud_provider) {
    $cloud_args = empty($cloud_config) ? {
      true    => ["cloud-provider: ${cloud_provider}"],
      default => ["cloud-provider: ${cloud_provider}", "cloud-config: ${cloud_config}"],
    }
    $kubelet_merged_extra_arguments = concat($kubelet_extra_arguments, $cloud_args)
  }
  else {
    $kubelet_merged_extra_arguments = $kubelet_extra_arguments
}
```
Otherwise the template is not going to include the option:
```
  kubeletExtraArgs:
    <%- @kubelet_merged_extra_arguments.each do |arg| -%>
    <%= arg %>
    <%- end -%>
<%- end -%>
```

## How to reproduce it?

Create a controller and a worker node using any valid argument for kubelet_extra_arguments and the result for the controller node should differ from the worker.

## Versions:

```
$ puppet --version
5.5.10

$ kubectl version
Client Version: version.Info{Major:"1", Minor:"13", GitVersion:"v1.13.4", GitCommit:"c27b913fddd1a6c480c229191a087698aa92f0b1", GitTreeState:"clean", BuildDate:"2019-02-28T13:37:52Z", GoVersion:"go1.11.5", Compiler:"gc", Platform:"linux/amd64"}
Server Version: version.Info{Major:"1", Minor:"13", GitVersion:"v1.13.4", GitCommit:"c27b913fddd1a6c480c229191a087698aa92f0b1", GitTreeState:"clean", BuildDate:"2019-02-28T13:30:26Z", GoVersion:"go1.11.5", Compiler:"gc", Platform:"linux/amd64"}

$ docker version OR crictl version
Docker version 1.11.2, build b9f10c9

$ facter os

{
  architecture => "amd64",
  distro => {
    codename => "stretch",
    description => "Debian GNU/Linux 9.5 (stretch)",
    id => "Debian",
    release => {
      full => "9.5",
      major => "9",
      minor => "5"
    }
  },
  family => "Debian",
  hardware => "x86_64",
  name => "Debian",
  release => {
    full => "9.5",
    major => "9",
    minor => "5"
  },
  selinux => {
    enabled => false
  }
}

$ puppet module list
```

## Logs:
```
Paste any relevant logs from a puppet run, kubernetes and syslog/messages
```

**Comment by davejrt** [comment]
Thank you for completing the template with the relevant information. We'll look into this and report back on this issue with more information once we've triaged the issue.

**Comment by davejrt** [comment]
This was fixed in #259 and went out this morning as part of version 3.3.0. 

Would you mind testing with the latest version and reporting back with the results?

**Comment by puetzp** [comment]
Works as expected now. Thank you!
